var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -234,7 +241,7 @@ class Llvm(CMakePackage, CudaPackage):
     depends_on("hwloc")
     depends_on("hwloc@2.0.1:", when="@9:")
     depends_on("elf", when="+cuda")  # libomptarget
-    depends_on("libffi", when="+cuda")  # libomptarget
+    depends_on("libffi", when="+libomptarget")  # libomptarget
```

Comment:
Since `+cuda` is valid for versions up to `@15` -- wouldn't we want to keep `libffi` as a dependency in those cases?

Reply:
`libffi` is needed for OpenMP offload to host (x86, ppc64le, ...). It is needed because of `+libomptarget` not `+cuda`. `+libomptarget` applies to all version of llvm with libomptarget included (initial commit in 2017 way before llvm 15).